### PR TITLE
LDAP-340: Upgrade apacheds to 1.5.7

### PR DESCRIPTION
**Pull request by eddumelendez**

### CODE REVIEW

File: test-support/src/main/java/org/springframework/ldap/test/EmbeddedLdapServer.java [hunk 1]
```diff
@@ -88,4 +107,49 @@ public void shutdown() throws Exception {
 
         FileUtils.deleteDirectory(workingDirectory);
     }
+
```
Comment (rwinch) [inline test-support/src/main/java/org/springframework/ldap/test/EmbeddedLdapServer.java]: The changes below seem fairly drastic for updating the apache ds version. Is there a reason they were necessary?

Comment (eddumelendez) [inline test-support/src/main/java/org/springframework/ldap/test/EmbeddedLdapServer.java]: Sorry for the delay, no reason to do that. That line can be omitted.

Comment (rwinch) [inline test-support/src/main/java/org/springframework/ldap/test/EmbeddedLdapServer.java]: Any chance you can update the PR to only include the changes that are necessary?

**Comment by eddumelendez** [comment]
@rwinch PR updated

**Comment by rwinch** [comment]
Thanks for the update. I commented inline.

**Comment by eddumelendez** [comment]
@rwinch sorry my mistake, i have updated the PR

File: test/integration-tests/src/test/java/org/springframework/ldap/itest/LdapTemplateRenameITest.java [hunk 2]
```diff
@@ -1,103 +1,102 @@
-/*
```
Comment (rwinch) [inline test/integration-tests/src/test/java/org/springframework/ldap/itest/LdapTemplateRenameITest.java]: I'm guessing this does not need to be in the commit either?

**Comment by rwinch** [comment]
@eddumelendez Thanks for the fast turnaround. We are almost there I think :) I commented inline again.

**Comment by eddumelendez** [comment]
done :)

File: test-support/src/main/java/org/springframework/ldap/test/EmbeddedLdapServer.java [hunk 3]
```diff
@@ -85,7 +104,50 @@ public static EmbeddedLdapServer newEmbeddedServer(String defaultPartitionName,
     public void shutdown() throws Exception {
         ldapServer.stop();
         directoryService.shutdown();
-
-        FileUtils.deleteDirectory(workingDirectory);
     }
+
+	private static void initSchemaPartition(DefaultDirectoryService directoryService)
```
Comment (rwinch) [inline test-support/src/main/java/org/springframework/ldap/test/EmbeddedLdapServer.java]: Sorry...I think this is the last thing.

This is likely due to my own ignorance, but is there a reason we need to modify the code for this update? Is this required to update to apacheds 1.5.7 or is it a separate goal (in which case it might be best as a distinct issue)?

**Comment by rwinch** [comment]
Wow...incredible turnaround :) I have one more comment.

**Comment by eddumelendez** [comment]
Changes were based on this example. I will take a look later to answer your
question.

https://svn.apache.org/repos/asf/directory/documentation/samples/trunk/embedded-sample/src/main/java/org/apache/directory/seserver/EmbeddedADSVer157.java

On Friday, 4 December 2015, Rob Winch <notifications@github.com> wrote:

> Wow...incredible turnaround :) I have one more comment.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/spring-projects/spring-ldap/pull/30#issuecomment-162067511>
> .
>


-- 
*Eddú Meléndez Gonzales*
*eddu.melendez@gmail.com* <eddu.melendez@gmail.com>
*@eddumelendez*

**Comment by rwinch** [comment]
Thanks! Looking forward to your findings

**Comment by eddumelendez** [comment]
I have fixed the tests.

File: test-support/src/main/java/org/springframework/ldap/test/EmbeddedLdapServer.java (hunk 3 above)
Comment (eddumelendez) [inline test-support/src/main/java/org/springframework/ldap/test/EmbeddedLdapServer.java]: This is required, without it ldap can not initialize. This method initializes the schema manager and add the schema partition to directory service.

Comment (rwinch) [inline test-support/src/main/java/org/springframework/ldap/test/EmbeddedLdapServer.java]: So they made a non-passive change? If so, we might need to include this in a non point release

File: test/integration-tests/src/test/java/org/springframework/ldap/itest/InvalidBackslashITest.java [hunk 4]
```diff
@@ -56,7 +56,6 @@ public void prepareTestedInstance() throws Exception {
 		adapter.setAttributeValue("sn", "Person6");
 		adapter.setAttributeValue("description", "Some description");
 
-		tested.unbind(DN);
```
Comment (rwinch) [inline test/integration-tests/src/test/java/org/springframework/ldap/itest/InvalidBackslashITest.java]: Why was this removed?

File: test/integration-tests/src/test/java/org/springframework/ldap/itest/LdapTemplateRenameITest.java [hunk 5]
```diff
@@ -63,7 +63,6 @@ public void prepareTestedInstance() throws Exception {
 	@After
 	public void cleanup() throws Exception {
 		tested.unbind(NEWDN);
-		tested.unbind(DN);
```
Comment (rwinch) [inline test/integration-tests/src/test/java/org/springframework/ldap/itest/LdapTemplateRenameITest.java]: Why was this removed?

**Comment by rwinch** [comment]
@eddumelendez Unfortunately the changes are causing failures in Spring IO Platform tests. Spring IO Platform uses Apache DS 1.5.5. Running the tests with the applied changes I get:

```
:spring-ldap-odm:springIoJdk7Test

org.springframework.ldap.odm.test.TestLdap > classMethod FAILED
    java.lang.NoClassDefFoundError at TestLdap.java:140
        Caused by: java.lang.ClassNotFoundException at TestLdap.java:140

org.springframework.ldap.odm.test.TestSchemaToJava > classMethod FAILED
    java.lang.NoClassDefFoundError at TestSchemaToJava.java:73
        Caused by: java.lang.ClassNotFoundException at TestSchemaToJava.java:73

org.springframework.ldap.odm.test.TestSchemaViewer > classMethod FAILED
    java.lang.NoClassDefFoundError at TestSchemaViewer.java:62
        Caused by: java.lang.ClassNotFoundException at TestSchemaViewer.java:62

9 tests completed, 3 failed
:spring-ldap-odm:springIoJdk7Test FAILED

FAILURE: Build failed with an exception.
```

So this means we would need to get Apache DS updated in Spring IO Platform first. Is there a concrete gain you are getting from updating ApacheDS? This will help when we create an issue with Spring IO Platform to update the version.

File: test/integration-tests/src/test/java/org/springframework/ldap/itest/InvalidBackslashITest.java (hunk 4 above)
Comment (eddumelendez) [inline test/integration-tests/src/test/java/org/springframework/ldap/itest/InvalidBackslashITest.java]: Throws `org.springframework.ldap.NameNotFoundException`

**Comment by pivotal-issuemaster** [comment]
@eddumelendez Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring)!

**Comment by pivotal-issuemaster** [comment]
@eddumelendez Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-ldap&pullRequestId=30)!

**Comment by rwinch** [comment]
I'm closing this as it breaks backwards compatability
